src/providers/ProfileModalProvider.js

```diff
@@ -2,26 +2,28 @@ import React, { createContext } from 'react';
 import { Text, TouchableOpacity, View } from 'react-native';
 import { User as Friend } from '../models/User';
 import ProfileModal from '../components/shared/ProfileModal';
+import { ProfileModalContext } from '../contexts/ProfileModalContext';
 
-const ProfileModalContext = createContext();
+const ProfileModalConsumer = ProfileModalContext.Consumer;
 
-export const ProfileModalConsumer = ProfileModalContext.Consumer;
-
-export class ProfileModalProvider extends React.Component {
+class ProfileModalProvider extends React.Component {
   static modal: ProfileModal;
 
-  state = {
-    user: null,
-  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: null,
+    };
+  }
 
   actions = {
     setModal: (v) => {
       this.modal = v;
     },
     showModal: (user: Friend, deleteMode?: boolean) => {
       this.setState({ user }, () => {
-        this.modal.setUser(user);
         if (this.modal) {
+          this.modal.setUser(user);
```

Comment:
this line had been placed above `if (this.modal)` on previous commit, it looks weird when `this.modal` is **undefined** or **null**. so I put into `if` scope.